test/performance/perf_test.py

```diff
@@ -429,11 +457,141 @@ def do_task(self):
         self.corpus.insert_many(self.documents, ordered=True)
 
 
+class TestSmallDocClientBulkInsert(SmallDocInsertTest, unittest.TestCase):
+    @client_context.require_version_min(8, 0, 0, -24)
+    def setUp(self):
+        super().setUp()
+        self.models = []
+        for doc in self.documents:
+            self.models.append(InsertOne(namespace="perftest.corpus", document=doc.copy()))
+
+    @client_context.require_version_min(8, 0, 0, -24)
+    def do_task(self):
+        self.client.bulk_write(self.models, ordered=True)
+
+
+class TestSmallDocBulkMixedOps(SmallDocMixedTest, unittest.TestCase):
+    def setUp(self):
+        super().setUp()
+        self.models = []
+        for doc in self.documents:
+            self.models.append(InsertOne(document=doc.copy()))
+            self.models.append(ReplaceOne(filter={}, replacement=doc.copy(), upsert=True))  # type: ignore[arg-type]
+            self.models.append(DeleteMany(filter={}))  # type: ignore[arg-type]
+
+    def do_task(self):
+        self.corpus.bulk_write(self.models, ordered=True)
+
+
+class TestSmallDocClientBulkMixedOps(SmallDocMixedTest, unittest.TestCase):
+    @client_context.require_version_min(8, 0, 0, -24)
+    def setUp(self):
+        super().setUp()
+        self.models = []
+        for doc in self.documents:
+            self.models.append(InsertOne(namespace="perftest.corpus", document=doc.copy()))
+            self.models.append(
+                ReplaceOne(
+                    namespace="perftest.corpus", filter={}, replacement=doc.copy(), upsert=True
+                )  # type: ignore[arg-type]
+            )
+            self.models.append(DeleteMany(namespace="perftest.corpus", filter={}))  # type: ignore[arg-type]
+
+    @client_context.require_version_min(8, 0, 0, -24)
+    def do_task(self):
+        self.client.bulk_write(self.models, ordered=True)
+
+
+class TestSmallDocMultiNamespaceBulkInsert(SmallDocMixedTest, unittest.TestCase):
```

Comment:
Is it useful to benchmark bulk_write with 2 namespaces?

Reply:
For the MultiNamespace cases I was curious to compare the two bulk_writes, since coll.bulk_write would have to make 2 server commands while client.bulk_write needs to make only 1. But yeah I'm not sure if it's that useful to benchmark!

Reply:
Let's remove the MultiNamespace ones. The main purpose of this benchmark should be to test the CPU perf of pymongo's ability to encode the bulk write commands. Mixing the namespace doesn't really change much there.